### PR TITLE
remove no-gen-bytecode flag

### LINKED ISSUE #43751: Re-enable --gen-bytecode in gen_snapshot once crashes are resolved.
The are currently several issues with the gen-bytecode option that required disabling it entirely. Until these are fixed and verified to work in google3, we probably shouldn't turn it back on.

See google3 issue: b/43536147 and b/143399169

cc @alexmarkov

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

The fix for this has rolled into flutter.

Fixes https://github.com/flutter/flutter/issues/43751

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/compile_batch_test.dart [hunk 1]
```diff
@@ -66,30 +66,6 @@ void main() {
...
-    when(mockFrontendServer.stdout)
-      .thenAnswer((Invocation invocation) => Stream<List<int>>.fromFuture(
-        Future<List<int>>.value(utf8.encode(
-          'result abc\nline1\nline2\nabc\nabc /path/to/main.dart.dill 0'
-        ))
-      ));
-    final KernelCompiler kernelCompiler = await kernelCompilerFactory.create(null);
-    await kernelCompiler.compile(sdkRoot: '/path/to/sdkroot',
-      mainPath: '/path/to/main.dart',
-      buildMode: BuildMode.release,
```
Comment (alexmarkov) [inline packages/flutter_tools/test/general.shard/compile_batch_test.dart]: Can we keep the test for compiling in release mode? We can check that other flags (like --aot) are properly passed.

**Review by alexmarkov — APPROVED** [review]
LGTM with suggestion.

File: packages/flutter_tools/test/general.shard/compile_batch_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/compile_batch_test.dart]: Done

Comment (alexmarkov) [reply packages/flutter_tools/test/general.shard/compile_batch_test.dart]: Thanks! Even more lgtm!

**Comment by jonahwilliams** [comment]
From talking offline with @zanderso, we're going to hold off on landing this until we confirm that --no-gen-bytecode can be removed from google3

**Comment by jonahwilliams** [comment]
Crash from a linux build:

```
[gradle_plugin_light_apk_test] [STDOUT] ../../third_party/dart/runtime/vm/object.cc: 3700: error: Unable to use class Library:'dart:core' Class: _List@0150898 which is not loaded yet.
[gradle_plugin_light_apk_test] [STDOUT] version=2.6.0-dev.8.2.flutter-f30b494035 (Wed Oct 30 12:25:54 2019 +0000) on "linux_simarm"
[gradle_plugin_light_apk_test] [STDOUT] thread=6548, isolate=isolate(0x123a540)
[gradle_plugin_light_apk_test] [STDOUT]   pc 0x00000000004e363c fp 0x00007ffda94ae270 /tmp/flutter sdk/bin/cache/artifacts/engine/android-arm-profile/linux-x64/gen_snapshot+0x2e363c
[gradle_plugin_light_apk_test] [STDOUT] -- End of DumpStackTrace
[gradle_plugin_light_apk_test] [STDOUT] Dart snapshot generator failed with exit code -6
[gradle_plugin_light_apk_test] [STDOUT] Exception: AOT snapshotter exited with code -6
[gradle_plugin_light_apk_test] [STDOUT] #0      AotElfBase.build 
...

**Comment by alexmarkov** [comment]
For some reason, we're passing `--no-link-platform` to front-end server when building dill file in profile mode. We don't pass this option in release mode. For AOT this option doesn't make sense and apparently harmful in case of bytecode (we literally do not include platform classes including _List from dart:core into dill file). Without this option flutter gallery compiles fine in profile mode.

File: packages/flutter_tools/test/general.shard/compile_batch_test.dart [hunk 2]
```diff
@@ -65,8 +65,40 @@ void main() {
...
+    when(mockFrontendServer.stdout)
+      .thenAnswer((Invocation invocation) => Stream<List<int>>.fromFuture(
+        Future<List<int>>.value(utf8.encode(
+          'result abc\nline1\nline2\nabc\nabc /path/to/main.dart.dill 0'
+        ))
+      ));
+    final KernelCompiler kernelCompiler = await kernelCompilerFactory.create(null);
+    await kernelCompiler.compile(sdkRoot: '/path/to/sdkroot',
+      mainPath: '/path/to/main.dart',
+      buildMode: BuildMode.release,
```
Comment (alexmarkov) [inline packages/flutter_tools/test/general.shard/compile_batch_test.dart]: BuildMode.profile?

**Comment by zanderso** [comment]
ping @jonahwilliams to check flag status w.r.t. the rolls.

**Comment by jonahwilliams** [comment]
Confirmed this has landed

**Review by alexmarkov — APPROVED** [review]
Still lgtm.

**Pull request merged**